app/models/portfolio_item.rb

```diff
@@ -24,4 +27,12 @@ def resolved_workflow_refs
   def item_workflow_ref
     [self, portfolio].detect(&:workflow_ref)&.workflow_ref
   end
+
+  def validate_workflow
+    raise Catalog::InvalidParameter if workflow_ref.blank?
```

Comment:
Per https://apidock.com/rails/ActiveRecord/Validations/ClassMethods/validate it is expected to add errors to a errors key when calling a method via `validate`. I believe they should otherwise return a boolean, though that may be a common procedure and not a requirement

Reply:
Yep you're right, rather than throwing an exception I just needed to do `throw :abort` after adding an error. Updated it to follow the correct convention.